src/Engine/Utils.java

```diff
@@ -177,17 +177,17 @@ public static ArrayList<XYCoord> findFloodFillArea(XYCoord start, Unit mover, Ar
 
   public static boolean isPathValid(Unit unit, GamePath path, GameMap map, boolean includeOccupiedSpaces)
   {
-    return isPathValid(new XYCoord(unit), unit, unit.CO.army, unit.getMoveFunctor(), Math.min(unit.getMovePower(map), unit.fuel), path, map, includeOccupiedSpaces);
+    return isPathValid(unit, unit.CO.army, unit.getMoveFunctor(), Math.min(unit.getMovePower(map), unit.fuel), path, map, includeOccupiedSpaces);
   }
-  public static boolean isPathValid(XYCoord start, Unit mover, Army team, MoveType fff, int initialFillPower, GamePath path, GameMap map, boolean includeOccupiedSpaces)
+  public static boolean isPathValid(Unit mover, Army team, MoveType fff, int initialFillPower, GamePath path, GameMap map, boolean includeOccupiedSpaces)
   {
     if( (null == path) || (null == fff) )
     {
       return false;
     }
 
     // Make sure the first waypoint is under the Unit.
-    if( path.getPathLength() <= 0 || !path.getWaypoint(0).GetCoordinates().equals(start) )
+    if( path.getPathLength() <= 0 )
```

Comment:
Comment out of date

Reply:
Also won't this check fail now for actions that a unit performs in situ? Units still "move" on a length-1 path to its current location before performing the action.

Reply:
Nope. A length-zero (or less) path is totally invalid, since the first point is the unit's starting position.